sygnal/sygnal.py

```diff
@@ -86,6 +85,11 @@ def __init__(self, config, custom_reactor, tracer=opentracing.tracer):
                 "name": "sqlite3",
                 "args": {"dbfile": config["db"]["dbfile"]},
             }
+        elif config.get("database") is None:
+            config["database"] = {
+                "name": "sqlite3",
+                "args": {"dbfile": "sygnal.db"},
+            }
```

Comment:
How is this different from what you removed?

Reply:
`merge_left_with_defaults` (in the `if __name__ == '__main__'`) statement would do a deep merge and  copy subkeys if the parent is present, which would cause `dbfile` to appear even if `args` was specified.

The change ensures `database` is only defined if it's not already defined.

Reply:
Right, makes sense.